tex/pgfcirctripoles.tex

```diff
@@ -136,421 +136,360 @@
-		\pgf@y=.5\pgf@y
-		\pgf@y=\pgfkeysvalueof{/tikz/circuitikz/tripoles/american #1 port/input height}\pgf@y
-		\pgf@x=-\pgfkeysvalueof{/tikz/circuitikz/tripoles/american #1 port/width}\pgf@circ@Rlen
-		\pgf@x=.5\pgf@x
-	  }	  
-	  \anchor{in 1}{
-		\inOne
-	  }
-	  \anchor{in}{
-		\inOne
-	  }	  
-	  \anchor{in 2}{
-		\inOne
-		\pgf@y=-\pgf@y
-	  }
-	  \anchor{out}{
-		\northwest
-		\pgf@y=0pt
-		\pgf@x=-\pgf@x
-	  }
-  	  \anchor{center}{
-		\northwest
-		\pgf@y=0pt
-		\pgf@x=-.8\pgf@x  
-	  }
-	  \anchor{east}{
-	  	\left
-	  	\pgf@x=-.8\pgf@x  
-	  }
-	  \anchor{west}{
-	  	\left
-	  }
-	  \anchor{south}{
-		\northwest
-		\pgf@x=0pt
-		\pgf@y=-\pgf@y
-	  }
-	  \anchor{north}{
-		\northwest
-		\pgf@x=0pt
-	  }
-	  \anchor{south west}{
-		\northwest
-		\pgf@y=-\pgf@y
-	  }
-	  \anchor{north east}{
-		\northwest
-		\pgf@x=-\pgf@x
-	  }
-	  \anchor{north west}{
-		\northwest
-	  }
-	  \anchor{south east}{
-		\northwest
-		\pgf@x=-\pgf@x
-		\pgf@y=-\pgf@y
-	  }	  
-	  \backgroundpath{			
-			\pgfsetcolor{\pgfkeysvalueof{/tikz/circuitikz/color}}	
-			
-			
-			\northwest
-			\pgf@circ@res@up = \pgf@y 
-			\pgf@circ@res@down = -\pgf@y
-			\pgf@circ@res@right = -\pgf@x
-			\pgf@circ@res@left = \pgf@x
-			
-			#2
-	  
-	  }
-	}
-}
 
+%%% added 
+
+\newcount{\pgf@circ@res@count}% reserve global register
+
+\newcommand{\pgf@circ@logicport@input}[1]% #1 = \pgfmathcounter
+{%
+  \pgfextracty{\pgf@circ@res@up}{\northeast}%
+  \step
+  \pgf@circ@res@step=\dimexpr 2\pgf@y -2\pgf@circ@res@up\relax
+  \advance\pgf@y by -#1\pgf@circ@res@step\relax
+}%
+
+%%% american
+\long\def\pgfcircdeclarelogicport#1#2{%
+    \pgfdeclareshape{american #1 port}%
+    {%
+    \savedmacro\resize{% automatic
+      \pgf@circ@res@up = \pgfkeysvalueof{/tikz/circuitikz/tripoles/american #1 port/height}\pgf@circ@Rlen
+      \pgf@circ@res@up = .5\pgf@circ@res@up
+      \pgf@circ@res@down = -\pgf@circ@res@up
+      \pgf@circ@res@right = \pgfkeysvalueof{/tikz/circuitikz/tripoles/american #1 port/width}\pgf@circ@Rlen
+      \pgf@circ@res@right = .5\pgf@circ@res@right
+      \pgf@circ@res@left = -\pgf@circ@res@right
+    }%
+    \savedmacro\inputs{% get number of inputs
+      \pgf@circ@res@count=\pgfkeysvalueof{/tikz/number inputs}\relax%
+      \ifnum\pgf@circ@res@count=0
+        \pgf@circ@res@count=\pgfkeysvalueof{/tikz/circuitikz/tripoles/american #1 port/inputs}\relax%
+      \fi
+      \ifnum\pgf@circ@res@count<2 \pgf@circ@res@count=2\fi
+      \ifnum\pgf@circ@res@count>16 \pgf@circ@res@count=16\fi
+      \def\inputs{\the\pgf@circ@res@count}%
+    }%
+    \savedanchor\step{% 1/2 gap at edges
+      \pgf@circ@res@step = \pgfkeysvalueof{/tikz/circuitikz/tripoles/american #1 port/height}\pgf@circ@Rlen
+      \divide\pgf@circ@res@step by \pgf@circ@res@count
+      \pgfpoint{\pgf@circ@res@left}{\dimexpr\pgf@circ@res@up+0.5\pgf@circ@res@step}%
+    }%
+      \savedanchor\northeast{\pgfpoint{\pgf@circ@res@right}{\pgf@circ@res@up}}
```

Comment:
Probably spaces should be equal in all lines…

```suggestion
    \savedanchor\northeast{\pgfpoint{\pgf@circ@res@right}{\pgf@circ@res@up}}
```

Reply:
Yes, I have an issue open for a big clean-up of spaces... accepted, of course.